src/features/auth/LoginPage.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import CardHeader from '@mui/material/CardHeader';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import InputAdornment from '@mui/material/InputAdornment';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import CircularProgress from '@mui/material/CircularProgress';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import "../../index.css";
+
+// Define the theme according to the style guide
+const theme = createTheme({
+  palette: {
+    primary: {
+      main: '#003366', // Navy Blue
+    },
+    secondary: {
+      main: '#87CEEB', // Sky Blue
+    },
+    background: {
+      default: '#F2F2F2', // Light Gray
+    },
+    text: {
+      primary: '#333333', // Dark Gray
+      secondary: '#666666', // Medium Gray
+    },
+  },
+  typography: {
+    fontFamily: [
+      'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'
+    ].join(','),
+  },
+});
+
+const LoginPage = () => {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const [showPassword, setShowPassword] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+
+  const onSubmit = async data => {
+    await sleep(2000);
+    console.log(JSON.stringify(data));
+  };
+
+  const togglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Box
+        display="flex"
+        justifyContent="center"
+        alignItems="center"
+        minHeight="100vh"
+        sx={{ bgcolor: 'background.default' }}
+      >
+        <Card sx={{ minWidth: 275, bgcolor: 'background.paper' }}>
+          <CardHeader title="Tech Resume Pro" subheader="Please login to continue." sx={{ color: 'text.primary' }} />
+          <CardContent>
+            <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
A lot of this logic is best pulled out into a hook (business logic) so that it can be tested a bit easier.